### PR TITLE
Add code style + unused vars checks

### PR DESCRIPTION
**Pull request by jeroenvandijk**
### Style check 
fails if fixes are found ([last run](https://github.com/jeroenvandijk/athens/runs/674359114#step:3:21))

### Unused vars check
fails if superfluous vars are found ([last run](https://github.com/jeroenvandijk/athens/runs/674359202#step:4:118))

### Next step
Fix the checks and/or tweaking the configuration. Probably good for beginners. Link to docs is in `script/style` and `script/carve`

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -61,6 +61,37 @@ jobs:
...
+        with:
+          fetch-depth: 1
+          submodules: 'true'
+          
+      - name: Style 
+        run: |
+          script/style
+                 
+  carve:
+    # ubuntu 18.04 comes with lein + java8 installed
```
Comment (tangjeff0) [inline .github/workflows/build.yml:80]: I'm using `java11`. Should we try to stick to one version?

**Comment by tangjeff0** [comment]
Also the `carve` script seems to fail, but not because of the check. It looks like it doesn't even run.

File: .github/workflows/build.yml (hunk 1 above)
Comment (jeroenvandijk) [reply .github/workflows/build.yml:80]: It's should not matter for most Clojure feature (time API's are in new jdks'). 

I haven't reinstalled java for a long time. Still running Java8. I'm not sure if I'm unique, but I wouldn't enforce it on the dev part, unless it becomes a problem.

On CI we might want to test against multiple version indeed. Or choose a version that we specifically support. I guess java11 makes sense, although I'm still focussed on Java8 myself. I don't see big issues here

**Comment by jeroenvandijk** [comment]
@tangjeff0 Regarding `carve` it's hidden in the output ```
({:filename "src/cljs/athens/core.cljs", :row 25, :col 1, :ns athens.core, :name init} {:filename "src/cljs/athens/db.cljs", :row 66, :col 1, :ns athens.db, :name help-url} {:filename "src/cljs/athens/db.cljs", :row 67, :col 1, :ns athens.db, :name ego-url} {:filename "src/cljs/athens/views.cljs", :row 50, :col 1, :ns athens.views, :name home-panel})
```

But could definitely improve the output here. Merging for now. Will fix it later

**Pull request merged**